src/utils/network_calculations/ybus_calculations.jl

```diff
@@ -186,3 +162,106 @@ function Ybus(sys::System; check_connectivity::Bool = true)
     fixed_admittances = get_components(FixedAdmittance, sys)
     return Ybus(branches, nodes, fixed_admittances; check_connectivity = check_connectivity)
 end
+
+"""
+Nodal incidence matrix (Adj) is an N x N matrix describing a power system with N buses. It represents the directed connectivity of the buses in a power system.
+
+The Adj Struct is indexed using the Bus Numbers, no need for them to be sequential
+"""
+struct Adj{Ax, L <: NTuple{2, Dict}} <: PowerNetworkMatrix{Int}
+    data::SparseArrays.SparseMatrixCSC{Int, Int}
+    axes::Ax
+    lookup::L
+end
+
+"""
+Builds a Adj from a collection of buses and branches. The return is a Adj Array indexed with the bus numbers and the branch names.
+
+# Keyword arguments
+- `check_connectivity::Bool`: Checks connectivity of the network using Goderya's algorithm
+"""
+function Adj(branches, nodes; check_connectivity::Bool = true)
```

Comment:
we should abstract the code in the PTDF matrix to do the same and have it as an internal method and avoid code duplication.

Reply:
Do you mean replace [this code](https://github.com/NREL-SIIP/PowerSystems.jl/blob/master/src/utils/network_calculations/ptdf_calculations.jl#L34-L51) with `Adj(branches, nodes)`? If so, they are not quite the same thing. The PTDF needs an incidence matrix (N x B), and `Adj` creates an adjacency matrix (N x N).

Reply:
@jd-lara, I made an attempt at minimizing code duplication here. My best effort netted a 4 line reduction in code. So, I'm inclined to leave it as is for readability/simplicity.